### PR TITLE
docstring parser

### PR DESCRIPTION
**Pull request by de-code**
part of #15

### CODE REVIEW

**Comment by de-code** [comment]
Current implementation:

```console
$ python -m r2t2 --static --docstring examples/docstring_reference.py 
Referenced in: my_great_function
Source file: examples/docstring_reference.py
Line: 1
        [1] automatically parsed from docstring - 10.5281/zenodo.1185316
```

**Comment by de-code** [comment]
Perhaps based on @dalonsoa change, it is expecting the reference text to be the DOI URL. I shall add the URL prefix in that case.

**Comment by de-code** [comment]
It is now adding the DOI URL prefix:

```console
$ python -m r2t2 --static --docstring examples/docstring_reference.py 
Referenced in: my_great_function
Source file: examples/docstring_reference.py
Line: 1
        [1] automatically parsed from docstring - https://doi.org/10.5281/zenodo.1185316
```

File: tests/test_docstring_parser.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    iter_extract_docstring_from_files
+)
+
+
+DOC_STRING_LINE_1 = 'the docstring line 1'
+DOC_STRING_LINE_2 = 'the docstring line 2'
+
+
+class TestIterExtractDocstringFromLines:
+    def _test_should_extract_no_docstrings_from_empty_file(self):
```
Comment (jezcope) [inline tests/test_docstring_parser.py:16]: Is the leading underscore to skip the test? If so, the canonical way to do this with pytest would be to decorate with `pytest.mark.skip` or `pytest.mark.xfail` depending on whether you expect it to currently fail or not.

**Review by jezcope — APPROVED** [review]
Looks good to me. :slightly_smiling_face:

File: tests/test_docstring_parser.py (hunk 1 above)
Comment (de-code) [reply tests/test_docstring_parser.py:16]: oops. That was meant to be there only temporarily. I shall fix that in another PR.

**Pull request merged**